extensions/blocks/podcast-player/podcast-player.php

```diff
@@ -0,0 +1,174 @@
+function render_block( $attributes, $content ) {
+
+	// Test for empty URLS.
+	if ( empty( $attributes['url'] ) ) {
+		return '<p>' . esc_html__( 'No Podcast URL provided. Please enter a valid Podcast RSS feed URL.', 'jetpack' ) . '</p>';
+	}
+
+	// Test for invalid URLs.
+	if ( ! wp_http_validate_url( $attributes['url'] ) ) {
+		return '<p>' . esc_html__( 'Your podcast URL is invalid and couldn\'t be embedded. Please double check your URL.', 'jetpack' ) . '</p>';
+	}
+
+	// Sanitize the URL.
+	$attributes['url'] = esc_url_raw( $attributes['url'] );
+
+	$track_list = get_track_list( $attributes['url'], $attributes['itemsToShow'] );
+
+	if ( is_wp_error( $track_list ) ) {
+		return '<p>' . esc_html( $track_list->get_error_message() ) . '</p>';
+	}
+
+	return render_player( $track_list, $attributes );
+}
+
+/**
+ * Renders the HTML for the Podcast player and tracklist.
+ *
+ * @param array $track_list the list of podcast tracks.
+ * @param array $attributes Array containing the Podcast Player block attributes.
+ * @return string the HTML for the podcast player.
+ */
+function render_player( $track_list, $attributes ) {
+
+	$player_data = array(
+		'type'         => 'audio',
+		// Don't pass strings to JSON, will be truthy in JS.
+		'tracklist'    => true,
+		'tracknumbers' => true,
+		'images'       => true,
+		'artists'      => true,
+		'tracks'       => $track_list,
+	);
+
+	$block_classname = Jetpack_Gutenberg::block_classes( FEATURE_NAME, $attributes );
+
+	// If there are no tracks (it is possible) then display appropriate user facing error message.
+	if ( empty( $track_list ) ) {
+		return '<p>' . esc_html__( 'No tracks available to play.', 'jetpack' ) . '</p>';
+	}
+
+	ob_start();
+
+	?>
+	<div class="<?php echo esc_attr( $block_classname ); ?>">
+		<?php // Placeholder: block markup is being handled in https://github.com/Automattic/jetpack/pull/14952. ?>
+		<script type="application/json" class="wp-playlist-script"><?php echo wp_json_encode( $player_data ); ?></script>
+	</div>
+	<?php
+	/*
+	* Enqueue necessary scripts and styles.
+	*/
+	Jetpack_Gutenberg::load_assets_as_required( 'podcast-player' );
+
+	return ob_get_clean();
+}
+
+/**
+ * Gets a list of tracks for the supplied RSS feed.
+ *
+ * @param string $feed the RSS feed to load and list tracks for.
+ * @param int    $quantity the number of tracks to return.
+ * @return array|WP_Error the feed's tracks or a error object.
+ */
+function get_track_list( $feed, $quantity = 5 ) {
+	if ( empty( $feed ) ) {
+		return new WP_Error( 'missing_feed', __( 'Podcast audio RSS feed missing.', 'jetpack' ) );
+	}
+
+	$rss = fetch_feed( $feed );
+
+	if ( is_wp_error( $rss ) ) {
+		return new WP_Error( 'invalid_url', __( 'Your podcast couldn\'t be embedded. Please double check your URL.', 'jetpack' ) );
+	}
+
+	if ( ! $rss->get_item_quantity() ) {
+		return new WP_Error( 'no_tracks', __( 'Podcast audio RSS feed has no tracks.', 'jetpack' ) );
+	}
+
+	$episodes = $rss->get_items( 0, $quantity );
```

Comment:
Since we have a maximum amount of items of 10 right now, I wonder if it would be an option to always fetch 10, and then only display the needed number of items. Would it speed things up on display a bit, especially when moving the range controller in the block sidebar?

Reply:
nice one. It will improve the user experience. 👍 
Updating the code.

Reply:
It should be handed [on this PR](https://github.com/Automattic/jetpack/pull/14952), actually.

Reply:
Going to do it, in the editor-canvas side, here https://github.com/Automattic/jetpack/pull/14973.

Reply:
If I'm not mistaken, the RSS feed includes as many episodes as the sender wants. `get_items()` will get a set amount of those, so I think it already does what we want it to do.
https://simplepie.org/api/class-SimplePie.html#_get_items